source/Ship.cpp

```diff
@@ -1302,7 +1302,7 @@ vector<string> Ship::FlightCheck() const
 			checks.emplace_back("afterburner only?");
 		if(!thrust && !afterburner)
 			checks.emplace_back("reverse only?");
-		if(!generation && !solar && !consuming)
+		if(energy <= battery)
```

Comment:
```suggestion
		if(generation + consuming <= 0 && !solar)
```
You are currently checking if the generated energy only lasts for a single frame, not whether the ship can generate energy.

Reply:
Your code is the same as mine (assuming you meant `if((generation + consuming <= 0.) && !solar)`) except where mine checks if `solar <= 0` not just `== 0`.
`energy` is defined above as `generation + consuming + solar + battery`. Everything is in per-frame values. Testing if `energy <= battery` would be the same as `generation + consuming + solar <= 0` except without any additional computation, just a `jle` assembly instruction since we already have the sum of all four and `battery` as a variable.

I have just tested both and they behave identically.